src/utils/src/lib.rs

```diff
@@ -18,3 +23,30 @@ macro_rules! debug {
         ()
     };
 }
+
+// -----------------------------------------------------------------------------
+/// Enum containing error types.
+pub enum Error {
+    Io(io::Error),
+    Path,
+}
+
+impl std::fmt::Display for Error {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(f, "Erorr: {}", (*self))
+    }
+}
+
+/// A safe wrapper around libc::mkfifo().
+pub fn mkfifo<P: AsRef<Path>>(fifo_path: P, mode: u32) -> result::Result<(), Error> {
+    let fifo_path_str = fifo_path.as_ref().to_str().ok_or(Error::Path)?;
+    let fifo_path_cstr = CString::new(fifo_path_str).unwrap();
```

Comment:
Why is this `unwrap` safe? Looking at the documentation of `Cstring::new`, it fails when the string contains 0 bytes in it:
https://doc.rust-lang.org/std/ffi/struct.CString.html#method.new